dpnp/backend/queue_sycl.cpp

```diff
@@ -118,7 +123,116 @@ void backend_sycl::backend_sycl_queue_init(QueueOptions selector)
     std::cout << "SYCL kernels link time: " << time_kernels_link.count() << " (sec.)\n" << std::endl;
 }
 
+void backend_sycl::backend_sycl_rng_engine_init()
+{
+    size_t seed;
+
+    if (engines[engine_type])
+    {
+        backend_sycl::destroy_rng_engine();
+    }
+
+    if(reseed)
+    {
+        seed = reseed_val;
+        reseed = false;
+    }
+    else
+    {
+        // TODO:
+        // choose engine as is in numpy
+        // seed number
+        // TODO:
+        // mem leak
+        seed = std::time(nullptr);
+    }
+    switch(engine_type)
+    {
+        case EngineOptions::ARS5:
+        {
+            engines[engine_type] = new mkl_rng::ars5(DPNP_QUEUE, seed);
```

Comment:
why do we need diffrent `engines`? Do we need them differently by some end-user function?
Can we use some single, preselected in sources, engine for all random functions?